File: maximum-level-sum-of-a-binary-tree/maximum-level-sum-of-a-binary-tree.py
# maximum level sum of binary tree
class Solution:
    def maxLevelSum(self, root: Optional[TreeNode]) -> int:
        mxSum, ans, level = float('-inf'), 0, 0
        q = collections.deque();
        q.append(root)
        while q:
            level += 1
            sumAtCurr = 0
            for _ in range(len(q)):
                node = q.popleft()
                sumAtCurr += node.val
                if node.left: q.append(node.left)
                if node.right: q.append(node.right)
            if mxSum < sumAtCurr:
                mxSum, ans = sumAtCurr, level
        return ans

File: maximum-level-sum-of-a-binary-tree/maximum-level-sum-of-a-binary-tree.cpp

class Solution {
public:
    int maxLevelSum(TreeNode* root) {
        int mxSum = INT_MIN, ans = 0, level = 0;
        queue<TreeNode*> q;
        q.push(root);
        while(!q.empty()){
            level++;
            int sum = 0;
            for(int i=q.size(); i>0; i--){
                TreeNode* node = q.front(); q.pop();
                sum += node->val;
                if(node->left != NULL) q.push(node->left);
                if(node->right != NULL) q.push(node->right);
            }
            if(mxSum < sum){
                mxSum = sum;
                ans = level;
            }
        }
        return ans;
    }
};